src/components/Pressable/GenericPressable/BaseGenericPressable.tsx

```diff
@@ -51,10 +36,30 @@ function GenericPressable(
     }: PressableProps,
     ref: ForwardedRef<View>,
 ) {
+    const styles = useThemeStyles();
     const {isExecuting, singleExecution} = useSingleExecution();
     const isScreenReaderActive = Accessibility.useScreenReaderStatus();
     const [hitSlop, onLayout] = Accessibility.useAutoHitSlop();
 
+    /**
+     * Returns the cursor style based on the state of Pressable
+     */
+    const cursorStyle = useMemo(
+        () =>
+            (isDisabled: boolean, isText: boolean): Pick<ViewStyle, 'cursor'> => {
+                if (isDisabled) {
+                    return styles.cursorDisabled;
+                }
+
+                if (isText) {
+                    return styles.cursorText;
+                }
+
+                return styles.cursorPointer;
+            },
+        [styles],
+    );
```

Comment:
```suggestion
    const cursorStyle = useMemo(() => {
        if (shouldUseDisabledCursor) {
            return styles.cursorDisabled;
        }

        if ([rest.accessibilityRole, rest.role].includes('text')) {
            return styles.cursorText;
        }

        return styles.cursorPointer;
    }, [styles, shouldUseDisabledCursor, rest.accessibilityRole, rest.role]);
```

cursorStyle does not need to be a function